pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -38,12 +39,33 @@ func NewAnnotationHistorian(annotations annotations.Repository, dashboards dashb
 func (h *AnnotationStateHistorian) RecordStatesAsync(ctx context.Context, rule *ngmodels.AlertRule, states []state.StateTransition) {
 	logger := h.log.FromContext(ctx)
 	// Build annotations before starting goroutine, to make sure all data is copied and won't mutate underneath us.
-	annotations := h.buildAnnotations(rule, states, logger)
+	annotations := buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
 	go h.recordAnnotationsSync(ctx, panel, annotations, logger)
 }
 
-func (h *AnnotationStateHistorian) buildAnnotations(rule *ngmodels.AlertRule, states []state.StateTransition, logger log.Logger) []annotations.Item {
+func (h *AnnotationStateHistorian) recordAnnotationsSync(ctx context.Context, panel *panelKey, annotations []annotations.Item, logger log.Logger) {
+	if panel != nil {
+		dashID, err := h.dashboards.getID(ctx, panel.orgID, panel.dashUID)
+		if err != nil {
+			logger.Error("Error getting dashboard for alert annotation", "dashboardUID", panel.dashUID, "error", err)
```

Comment:
Should we also return this error to the caller?

Reply:
No, this was intentional,
1. This function is the entry point to a goroutine
2. Since annotations are an auxiliary thing, we don't want errors to bubble up the stack. This is a case of "if it fails, log it and move on."

Reply:
I think we should not even `return` but continue with empty dashboard and panel IDs because otherwise we will break state history

Reply:
Sure, but this is completely orthogonal to this PR. I doubt this is the only issue we have of this nature, this code was simply moved inside the file and not touched.

I think we should do that in a separate PR, I'd rather not lump it in here, because it deserves its own tests just for that case.